public/app/ui/pages/detail/library.html

```diff
@@ -1,3 +1,209 @@
 <article id="library-detail" class="detail">
-    <h1>Library Record</h1>
+    <div class="type-header">Library Record</div>
```

Comment:
This is the key file.  As you can see there is a lot of logic in the template (`ko if: ...` and `ko foreach: ...`), and it's quite repetitive.  So I'll probably look at knockout components but just wanted to discuss it with you first.